poetry/installation/authenticator.py

```diff
@@ -61,8 +67,16 @@ def request(self, method: str, url: str, **kwargs: Any) -> requests.Response:
 
         proxies = kwargs.get("proxies", {})
         stream = kwargs.get("stream")
-        verify = kwargs.get("verify")
-        cert = kwargs.get("cert")
+
+        certs = self.get_certs_for_url(url)
+        verify = kwargs.get("verify", certs.get("verify"))
+        cert = kwargs.get("cert", certs.get("cert"))
```

Comment:
```suggestion
        verify = kwargs.get("verify") or certs.get("verify")
        cert = kwargs.get("cert") or certs.get("cert")
```

This should handle potential cases where `None` is explicitly passed through.

Reply:
Addressed in commit 26b4ce7731dea5c6ce7fb1980dae4a2b52e66ce3